upgrades/upgrade_testing_framework/steps/step_create_source_snapshot.py

```diff
@@ -0,0 +1,49 @@
+import json
+
+from upgrade_testing_framework.core.framework_step import FrameworkStep
+import upgrade_testing_framework.core.shell_interactions as shell
+
+class CreateSourceSnapshot(FrameworkStep):
+    """
+    This step creates the snapshot on the source cluster.
+    """
+
+    def _run(self):
+        """
+        The contents of this are quick/dirty.  Instead of raw curl commands we should probably be using the Python SDK
+        for Elasticsearch/OpenSearch.  At the very least, we should wrap the curl commands in our own pseudo-client.
+        """
+
+        # Get the state we need
+        shared_volume = self.state.shared_volume
+        source_cluster = self.state.source_cluster
+
+        # Begin the step body
+        self.logger.info("Creating snapshot of source cluster...")
+        port = source_cluster.rest_ports[0]
+
+        register_cmd = f"""curl -XPUT 'localhost:{port}/_snapshot/noldor' -H 'Content-Type: application/json' -d '{{
```

Comment:
Seems like we should be using a constant here for snapshot repo name and snapshot name

Reply:
Sure - assuming this is our "real" code.  It isn't, however, and will be ripped out/replaced in short order once we write our actual test code.  Think of this as prototype/placeholder/demo code that is meant solely to illustrate how the UTF is designed to work.